experimental/dds/tree2/src/test/feature-libraries/editable-tree-2/objectFactory.spec.ts

```diff
@@ -21,10 +21,15 @@ describe("SharedTreeObject factories", () => {
 		content: sb.number,
 	});
 
+	const childOptional = sb.struct("childOptional", {
+		content: sb.optional(sb.number),
+	});
+
 	const parent = sb.struct("parent", {
 		child: [childA],
 		poly: [childA, childB],
 		list: sb.fieldNode("list", sb.sequence(sb.number)),
+		optional: sb.optional(childOptional),
```

Comment:
Note that in addition to the type checking in the new test, this is also testing the compilation of the implicitly undefined `optional` in `initialTree`, since `initialTree` omits the property.